Source/Core/DolphinQt2/GameList/TableDelegate.cpp

```diff
@@ -52,24 +58,44 @@ void TableDelegate::paint(QPainter* painter, const QStyleOptionViewItem& option,
 
 QSize TableDelegate::sizeHint(const QStyleOptionViewItem& option, const QModelIndex& index) const
 {
+  const auto scale = DPI::GetScale();
   switch (index.column())
   {
   case GameListModel::COL_PLATFORM:
-    return Resources::GetPlatform(0).size();
+    return Resources::GetPlatform(0).size() / Resources::GetIconScale() * scale;
   case GameListModel::COL_COUNTRY:
-    return Resources::GetCountry(0).size();
+    return Resources::GetCountry(0).size() / Resources::GetIconScale() * scale;
   case GameListModel::COL_RATING:
-    return Resources::GetRating(0).size();
+    return Resources::GetRating(0).size() * scale;
   case GameListModel::COL_BANNER:
-    return NORMAL_BANNER_SIZE;
+    return NORMAL_BANNER_SIZE * scale;
   default:
     return QSize(0, 0);
   }
 }
 
-void TableDelegate::DrawPixmap(QPainter* painter, const QRect& rect, const QPixmap& pixmap) const
+void TableDelegate::DrawPixmap(QPainter* painter, const QRect& rect, const QPixmap& pixmap,
+                               const QSize& size) const
 {
   // We don't want to stretch the pixmap out, so center it in the rect.
-  painter->drawPixmap(rect.left() + (rect.width() - pixmap.width()) / 2,
-                      rect.top() + (rect.height() - pixmap.height()) / 2, pixmap);
+  painter->drawPixmap(rect.left() + (rect.width() - size.width()) / 2,
+                      rect.top() + (rect.height() - size.height()) / 2, size.width(), size.height(),
+                      pixmap);
+}
+
+void TableDelegate::DrawPixmap(QPainter* painter, const QRect& rect, const QPixmap& pixmap,
+                               bool icon_scaled) const
+{
+  const auto scale = DPI::GetScale();
+
+  const auto width =
+      static_cast<double>(pixmap.width()) / (icon_scaled ? Resources::GetIconScale() : 1) * scale;
```

Comment:
Just do `pixmap.width() / pixmap.devicePixelRatio()`. Same comment as before about the chain of `GetScale()`, `GetIconScale()`, etc. that doesn't work.